src/Microsoft.ML.Data/Transforms/Normalizer.cs

```diff
@@ -251,19 +251,23 @@ private static VersionInfo GetVersionInfo()
                 loaderAssemblyName: typeof(NormalizerTransformer).Assembly.FullName);
         }
 
-        private class ColumnInfo
+        public sealed class ColumnInfo
         {
             public readonly string Input;
             public readonly string Output;
             public readonly ColumnType InputType;
-            public readonly IColumnFunction ColumnFunction;
+            public readonly NormalizerModelParametersBase ModelParameters;
+            internal readonly IColumnFunction ColumnFunction;
 
-            public ColumnInfo(string input, string output, ColumnType inputType, IColumnFunction columnFunction)
+            private ColumnInfo() { }
+
+            internal ColumnInfo(string input, string output, ColumnType inputType, IColumnFunction columnFunction)
             {
                 Input = input;
                 Output = output;
                 InputType = inputType;
                 ColumnFunction = columnFunction;
+                ModelParameters = columnFunction?.GetNormalizerModelParams();
```

Comment:
>columnFunction [](start = 34, length = 14)

when can it be null?

Reply:
It shouldn't but just in case 

---
In reply to: [234296040](https://github.com/dotnet/machinelearning/pull/1594#discussion_r234296040) [](ancestors = 234296040)

Reply:
well, I think you introduce more confusion with `?.` than without it.

---
In reply to: [234299210](https://github.com/dotnet/machinelearning/pull/1594#discussion_r234299210) [](ancestors = 234299210,234296040)